pkg/sql/plan/function/func_llm.go

```diff
@@ -0,0 +1,169 @@
+package function
+
+import (
+	"fmt"
+	"github.com/matrixorigin/matrixone/pkg/common/moerr"
+	"github.com/matrixorigin/matrixone/pkg/common/util"
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+	"io"
+)
+
+// Embedding function
+func EmbeddingOp(parameters []*vector.Vector, result vector.FunctionResultWrapper, proc *process.Process, length int, selectList *FunctionSelectList) error {
+	source := vector.GenerateFunctionStrParameter(parameters[0])
+	rs := vector.MustFunctionResult[types.Varlena](result)
```

Comment:
still hardcode ollama in the code. To create EmbeddingService, call NewEmbeddingService(modelStr).  Error out if modelStr not supported and only support "ollama" right now.